### ISSUE TITLE
aws_ssm_resource_data_sync - doesn't support update

### ISSUE THREAD

**Issue by egirard78** [terraform-providers/terraform-provider-aws#7351]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform version: 0.11.10
AWS provider: 1.57.0

### Affected Resource(s)

* aws_ssm_resource_data_sync

### Terraform Configuration Files

```hcl
provider "aws" {
  region      = "eu-west-1"
  profile     = "MyAWSProfile"
}

resource "aws_ssm_resource_data_sync" "data_sync" {
  name = "SSM-Sync-NewName"
  s3_destination = {
    bucket_name = "MySSMBucket"
    prefix      = "Inventory"
    region      = "eu-west-1"
  }
}
```

### Debug Output

Debug output related to the issue: [aws_ssm_resource_data_sync - Debug log](https://gist.github.com/egirard78/1cf0b04f23a6bf1ca6f189f68b36bf1b#file-gistfile1-txt).

### Panic Output

None.

### Expected Behavior

Recreate the aws_ssm_resource_data_sync when changing the S3 configuration of this resource.

### Actual Behavior

The following error message:

```shell
aws_ssm_resource_data_sync.data_sync: doesn't support update
```

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

None.

### References

None.

* #0000

**Comment by bflad** [comment]
Hi @egirard78 👋 Thanks for reporting this issue and sorry for the unexpected behavior! This is certainly an implementation bug with the resource's schema definition. Each of the attributes under the `s3_destination` attribute should also be marked as `ForceNew: true` as both the SSM service and subsequently the Terraform resource do not support updating this resource.

Normally this should be caught by the Terraform schema validation that occurs when starting the provider, but it appears we have some missing validation for situations involving children attributes like these. I'll take a note of this case here so we can get that validation added and prevent this issue in the future for all Terraform resources.

As a workaround to the current situation, you can use [`terraform taint`](https://www.terraform.io/docs/commands/taint.html) to manually trigger resource recreation with Terraform, e.g.

```sh
$ terraform taint aws_ssm_resource_data_sync.data_sync
```

**Comment by bflad** [comment]
The fix for having the resource automatically trigger resource updates when attributes inside the `s3_destination` configuration block are updated has been merged and will release with version 1.59.0 of the Terraform AWS Provider, likely middle of this week. 👍

**Comment by bflad** [comment]
This has been released in [version 1.59.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1590-february-14-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Comment by egirard78** [comment]
I confirm the issue is fixed in the 1.59 release. Thanks a lot!!
